includes/admin/class-wcmypa-admin.php

```diff
@@ -463,65 +481,139 @@ public function showOrderActions($order): void
             return;
         }
 
-        $shipping_country = WCX_Order::get_prop($order, 'shipping_country');
+        $shippingCountry = WCX_Order::get_prop($order, 'shipping_country');
 
-        if (! WCMP_Country_Codes::isAllowedDestination($shipping_country)) {
+        if (! WCMP_Country_Codes::isAllowedDestination($shippingCountry)) {
             return;
         }
 
-        $order_id = WCX_Order::get_id($order);
+        $listingActions = self::getListingActions($order);
+        $attributes     = self::getListingAttributes($order);
 
-        $baseUrl      = "admin-ajax.php?action=" . WCMP_Export::EXPORT;
-        $addShipments = WCMP_Export::ADD_SHIPMENTS;
-        $getLabels    = WCMP_Export::GET_LABELS;
-        $addReturn    = WCMP_Export::ADD_RETURN;
+        foreach ($listingActions as $data) {
+            self::renderAction(
+                $data['url'],
+                $data['alt'],
+                $data['img'],
+                $attributes
+            );
+        }
+    }
+
+    /**
+     * @param \WC_Order $order
+     *
+     * @return array|array[]
+     * @throws \Exception
+     */
+    public static function getListingActions(WC_Order $order): array
+    {
+        $orderId         = WCX_Order::get_id($order);
+        $shippingCountry = WCX_Order::get_prop($order, 'shipping_country');
+        $exportMode      = WCMYPA()->setting_collection->getByName(WCMYPA_Settings::SETTING_EXPORT_MODE);
+        $consignments    = self::get_order_shipments($order);
+        $listingActions  = self::getDefaultListingActions($orderId);
+
+        if (WCMP_Settings_Data::EXPORT_MODE_PPS === $exportMode) {
+            $metaPps        = get_post_meta($orderId, self::META_PPS);
+            $listingActions = self::updateExportButtonForPps($listingActions, $metaPps);
+        }
+
+        if (empty($consignments) || WCMP_Settings_Data::EXPORT_MODE_PPS === $exportMode) {
+            unset($listingActions[WCMP_Export::GET_LABELS]);
+        }
+
+        if (empty($consignments) || 'NL' !== $shippingCountry) {
```

Comment:
als je hier toch bent... vervangen door constante! WCMP_Data::base_country of iets dergelijks